buildstockbatch/gcp/gcp.py

```diff
@@ -72,13 +73,27 @@ def upload_directory_to_GCS(local_directory, bucket, prefix):
             local_filepath = pathlib.Path(dirpath, filename)
             string_paths.append(str(local_filepath.relative_to(local_dir_abs)))
 
-    transfer_manager.upload_many_from_filenames(
-        bucket,
-        string_paths,
-        source_directory=local_dir_abs,
-        blob_name_prefix=prefix,
-        raise_exception=True,
-    )
+    try:
+        transfer_manager.upload_many_from_filenames(
+            bucket,
+            string_paths,
+            source_directory=local_dir_abs,
+            blob_name_prefix=prefix,
+            raise_exception=True,
+            # Default chunk size is 40 MiB
+            blob_constructor_kwargs={
+                "chunk_size": chunk_size_mib * 1024 * 1024,
+            }
+            if chunk_size_mib
+            else None,
+        )
+    except requests.exceptions.ConnectionError:
+        logger.error("Error while uploading files to GCS bucket.")
+        logger.error(
```

Comment:
1. I think it'd be better for this to be just one log message.

2. Logging and rethrowing is somewhat of an anti-pattern, though I'm less sure about that for Python (but I think the principles still apply).
    * How that applies here, theoretically, is that the stack trace (which tells us whether this is a timeout error or something else) could be distant from this message about it. Admittedly, in this case there probably won't be any distance.

Given those, I think what you should do instead is raise a new exception (that contains this info) chained with the original exception.

Reply:
Good point - fixed.